backend/createStaffAndRooms.py

```diff
@@ -44,74 +44,81 @@ def search_ticket(ticket, guest_entries):
-            elem = {x.replace(' ', ''): v for x, v in elem.items()}
-            elem = {x: v.replace(' ', '') for x, v in elem.items() if type(v)==str}
-            dr.append(elem)
-
-    for elem in dr:
-        rooms.append(Room(name_take3=elem['Take3Name'],
-                          name_hotel=elem['HotelName'],
-                          number=elem['Number'],
-                          )
-                    )
-    list(map(lambda x: x.save(), rooms))
-
-
-def create_rooms_main(init_file =""):
-    dr = None
-    # {hotel_name: take3_name}
-    type_mapping = [
-           {"Hearing Accessible King": "Standard Room (1 King Bed)"},
-           {"Queen": "Standard Room (2 Queen Beds)"},
-           {"Executive Suite": "Knew Management Suite (1 King Bed)"},
-           {"Accessible Queen": "Standard Room (2 Queen Beds)"},
-           {"Hearing Accessible Queen": "Standard Room (2 Queen Beds)"},
-           {"King": "Standard Room (1 King Bed)"},
-           {"Smoking Executive Suite": "Knew Management Suite (1 King Bed)"},
-           {"Smoking Queen": "Standard Room (2 Queen Beds)"},
-           {"Smoking King": "Standard Room (1 King Bed)"},
-           {"Smoking Accessible Queen": "Standard Room (2 Queen Beds)"},
-           {"Lakeview King": "Lakeview Standard Room (1 King Bed)"},
-           {"Lakeview Queen": "Lakeview Standard Room (2 Queen Beds)"},
-           {"Accessible King": "Standard Room (1 King Bed)"},
-           {"2 Queen Accessible Sierra Suite": "Babyface Suite (2 Queen Beds)"},
-           {"2 Queen Sierra Suite": "Babyface Suite (2 Queen Beds)"},
-           {"Wedding Office": "IGNORE"},
-           {"Chapel": "IGNORE"},
-           {"1 King Sierra Suite": "Babyface Suite (1 King Bed)"},
-           {"Lakeview 1 King Sierra Suite": "Babyface Suite (1 King Bed)"},
-           {"Accessible 1 King Lakeview Sierra Suite": "Babyface Suite (1 King Bed)"},
-           {"Tahoe Suite": "Clavae Suite (1 King Bed)"},
-           {"Smoking 1 King Sierra Suite": "Babyface Suite (1 King Bed)"},
-           {"Smoking Tahoe Suite": "Clavae Suite (1 King Bed)"},
-           {"Smoking 2 Queen Sierra Suite": "Babyface Suite (2 Queen Beds)"},
-           {"Smoking Lakeview Queen": "Lakeview Standard Room (2 Queen Beds)"},
-    ]
-    rooms=[]
-    with open(init_file, "r") as f1:
-        dr = [elem for elem in DictReader(f1)]
-    for elem in dr:
-        #TODO(tb): omg this is big O off the charts. make it more efficient
-        for name in type_mapping:
-            if(elem["Room Type"] in name.keys()):
-                take3_name = name[elem["Room Type"]]
-
-        if(elem["ROOMBAHT"]=="R"):
-            rooms.append(Room(name_hotel=elem['Room Type'].lstrip(),
-                             number=elem['Room'].lstrip(),
-                             available=True,
-                             name_take3=take3_name
-                             )
-                       )
+    rooms_rows = []
+    types2023RoomList = {
```

Comment:
ten bux says i pull this out to a separate file once the dust has settled....

Reply:
yesssss